svetlana_kurakina_ndvi.py

```diff
@@ -0,0 +1,904 @@
+        # append site names, dates and mean ndvi to a list
+        all_ndvi.append([sitename, date, ndvi])
+
+# Construct data frame using list above
+ndvi = pd.DataFrame(all_ndvi, 
+                    columns = ['site', 'date', 'mean_ndvi'])
+
+# Convert string with date to a datetime object
+ndvi['date'] = pd.to_datetime(ndvi['date'])
+
+# Set dataframe index to a datetime
+ndvi.set_index('date', inplace=True)
+
+# Remove invalid ndvi values (i.e. remove rows with no values)
+ndvi_clean = ndvi[ndvi['mean_ndvi'] > 0]
+ndvi_clean
+
+
+# In[10]:
+
+
+# Last sanity check before creating your plot (10 points)
+
+# Ensure that you call your dataframe at the bottom of the cell above
+# and that it has columns called: mean_ndvi and site
+
+# Ensure the data is stored in a dataframe.
+student_ndvi_df = _
+
+df_points = 0
+
+if isinstance(student_ndvi_df, pd.DataFrame):
+    print('\u2705 Your data is stored in a DataFrame!')
+    df_points +=2
+else:
+    print('\u274C It appears your data is not stored in a DataFrame. ',
+          'To see what type of object your data is stored in, check its type with type(object)')
+
+# Check that dataframe contains the appropriate number of NAN values
+if student_ndvi_df.mean_ndvi.isna().sum() == 15:
+    print('\u2705 Correct number of masked data values!')
+    df_points +=2
+else:
+    print('\u274C The amount of null data in your dataframe is incorrect.')
+
+
+# Ensure that the date column is the index
+if isinstance(student_ndvi_df.index, pd.core.indexes.datetimes.DatetimeIndex):
+    print('\u2705 You have the index set to the date column!')
+    df_points +=3
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_df.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    df_points +=3
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Output for timer, # DO NOT MODIFY
+end_time = datetime.now()
+total_time = end_time - start_time
+print(
+    "Your total run time for processing the data was {0}.".format(total_time))
+
+print("\n \u27A1 You received {} out of 10 points for creating a dataframe.".format(
+    df_points))
+
+df_points
+
+
+# In[11]:
+
+
+# Add only the plot code to this cell
+
+# This is the final figure of mean NDVI
+# for both sites across the year
+# with data cleaned to deal with clouds
+
+# YOUR CODE HERE
+
+# Set a list of sites to loop through
+plot_sites = ['HARV', 'SJER']
+# Set a list of unique colors
+plot_colors = ['purple', 'green']
+
+# Set figure
+fig, ax = plt.subplots(figsize=(12, 6))
+
+# Loop through dataframe and plot NDVI for each site using assigned color
+for idx, asite in enumerate(plot_sites):
+    temp = ndvi_clean[ndvi_clean['site']==asite]
+    ax.plot(temp.index,
+            temp.mean_ndvi,
+            label=plot_sites[idx],
+            color = plot_colors[idx])
```

Comment:
I like this loop for plotting, a lot cleaner than the one I used!

Reply:
Thank you! I initially made a simpler loop, but it was assigning random color to my plot lines which was okay, but I wanted to be able to assign a specific color to each site. Thus I ended up wiht this solution, but it took me some googling and asking for advice from a person more experienced with plotting in python.